plugins/draco/io/DracoWriter.cpp

```diff
@@ -0,0 +1,457 @@
+        }
+    }
+    //red, green, and blue must all be specified if one of them is
+    if (m_userDimMap.find(Dimension::Id::Red) != m_userDimMap.end())
+    {
+        std::string r, g, b;
+        try {
+            r = m_userDimMap.at(Dimension::Id::Red);
+            g = m_userDimMap.at(Dimension::Id::Green);
+            b = m_userDimMap.at(Dimension::Id::Blue);
+        } catch(std::out_of_range e) {
+            throw pdal_error("Red, Green, and Blue dimensions must all be specified if one is.");
+        }
+        if ( r != g || g != b || r != b ) {
+            throw pdal_error("Red, Green, and Blue dimensions must be of the same type");
+        }
+    }
+    //normals x, y, z must all be specified if one of them is
+    if (m_userDimMap.find(Dimension::Id::NormalX) != m_userDimMap.end())
+    {
+        std::string nx, ny, nz;
+        try {
+            nx = m_userDimMap.at(Dimension::Id::NormalX);
+            ny = m_userDimMap.at(Dimension::Id::NormalY);
+            nz = m_userDimMap.at(Dimension::Id::NormalZ);
+        } catch(std::out_of_range e) {
+            throw pdal_error("NormalX, Y, and Z dimensions must all be specified if one is.");
+        }
+        if ( nx != ny || ny != nz || nx != nz ) {
+            throw pdal_error("NormalX, Y, and Z dimensions must be of the same type");
+        }
+    }
+    //make sure textures u, v, w are all the same, but it's possible there are
+    //only u and v, so make sure we know how many dimensions are in the file first
+    if (m_userDimMap.find(Dimension::Id::TextureU) != m_userDimMap.end())
+    {
+        std::string tu, tv, tw;
+        int texNum = m_dims.at(draco::GeometryAttribute::TEX_COORD);
+        try {
+            tu = m_userDimMap.at(Dimension::Id::TextureU);
+            tv = m_userDimMap.at(Dimension::Id::TextureV);
+            if (texNum == 3) tw = m_userDimMap.at(Dimension::Id::TextureW);
+            else tw = tv;
+        } catch(std::out_of_range e) {
+            throw pdal_error("TextureU, V, and W dimensions don't match the file.");
+        }
+        if ( tu != tv || tv != tw || tu != tw ) {
+            throw pdal_error("TextureU, V, and W dimensions must be of the same type");
+        }
+    }
+}
+
+void DracoWriter::ready(pdal::BasePointTable &table)
+{
+    auto layout = table.layout();
+
+    pdal::Dimension::IdList dimensions = layout->dims();
+
+    for (auto& dim : dimensions)
+    {
+        //create list of dimensions needed
+        int numComponents(1);
+        const auto it = dimMap.find(dim);
+        if (it != dimMap.end())
+        {
+            //get draco type associated with the current dimension in loop
+            draco::GeometryAttribute::Type dracoType = it->second;
+            if (!m_dims.count(dracoType)) m_dims[dracoType] = 1;
+            else ++m_dims.at(dracoType);
+        }
+        else
+        {
+            m_genericDims.push_back(dim);
+        }
```

Comment:
I unfortunately can't add the Draco attributes to the point cloud until it has been initialized, and the earliest I can do that is in the `write` method, so I create `m_genericDims` so that I can use it in `parseDimensions`

Reply:
Can you not just delay calling `set_num_points()` until `write()` and do all the rest earlier?

Reply:
I don't believe you can set attributes until after you've initialized the pointcloud. I was getting seg faults when I tried to do this.